### PR TITLE
UI: Update PostgreSQL database params

### PR DESCRIPTION
**Pull request by hellobontempo**
### Description
Added some missing params to PostgreSQL: 
- `tls_ca`
- `self_managed` (Enterprise only)
- `disable_escaping`
- `password_authentication`
- `auth_type`
-  `service_account_json`
-  `use_private_ip`
-  `private_key`
-  `tls_certificate`
Screenshot 2024-12-16 at 3 39 20 PM
Screenshot 2024-12-16 at 3 38 57 PM
Screenshot 2024-12-16 at 3 39 08 PM

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this fix needs to be backported, use the appropriate `backport/` label that matches the desired release branch. Note that in the CE repo, the latest release branch will look like `backport/x.x.x`, but older release branches will be `backport/ent/x.x.x+ent`.
    - [ ] **LTS**: If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: ui/tests/unit/models/database-connection-test.js
Comment (hellobontempo) [inline ui/tests/unit/models/database-connection-test.js:13]: This could be moved to a test helper file, but decided to hold off unless we plan to use it elsewhere?

I wrote this test coverage before adding the `_filterFields` function to the `database/connection` model

File: ui/app/models/database/connection.js
Comment (Monkeychip) [inline ui/app/models/database/connection.js:245]: I like this approach. nice.

File: ui/app/models/database/connection.js
Comment (hellobontempo) [inline ui/app/models/database/connection.js:205]: Is a future improvement, it might be good to have this field automatically toggle `verify_connection` but all of the databases use the same form, so this is a larger refactor.

Comment (Monkeychip) [reply ui/app/models/database/connection.js:205]: "If set," is used twice and thus reads a little funny. You could remove the first usage.

File: ui/app/models/database/connection.js
Comment (Monkeychip) [inline ui/app/models/database/connection.js:220]:
```suggestion
    // fields that are displayed for create and edit views regardless of the plugin type selected
    // does not include fields within a group as those fields are iterated over dynamically depending on what plugin the user has selected
```

File: ui/tests/unit/models/database-connection-test.js
Comment (Monkeychip) [inline ui/tests/unit/models/database-connection-test.js:12]:
```suggestion
// Ideal scenario would be that the API provides this schema so we don't have to manually write it :)
```

**Review by Monkeychip — APPROVED** [review]
Nice work! My suggestions are NITs so feel free to ignore if you don't want to ping me for another review.

File: ui/app/models/database/connection.js
Comment (hellobontempo) [reply ui/app/models/database/connection.js:205]: Thanks! Just copied this directly from the API docs 🙃

File: ui/app/models/database/connection.js
Comment (hellobontempo) [reply ui/app/models/database/connection.js:220]: I think that might be a little confusing, because below fields are added depending on the plugin type.

**Review by Monkeychip — APPROVED** [review]

**Pull request merged**
